### DISCUSSION TITLE
Geolocation (and general) tracking of qrcode searches

### DISCUSSION THREAD

**Discussion by Petrox** [inventree/InvenTree#5435] [Ideas]
I won't open issues on ideas out of the blue sky, I'll let you decide.

Some users might have multiple facilities and outdoor storages, maybe vehicles tools.
Sometimes assets go missing.
It would be very useful if the barcode/qrcode scans could be tracked back to see "last known location" (optionally with username)
That could give a hint where to look.

Unfortunately this feature needs to be implemented through the mobile app and the backend at the same time. Also it would require location access, that might trigger some users, they would also understand the context.

This feature could be also used with static smart gates (reading every code leaving the warehouse) or smart automatic systems like conveyor belts.

**Comment by SchrodingersGat** [comment]
@Petrox you might have to explain this one further. Each stock item already reports its last "known" location - I am not sure how the barcodes would come into it?

**Comment by Petrox** [comment]
Maybe it's in latest, but not in stable? I can't find where to see the last known location of a stock in my stable version either on web or mobile.

My idea was to track any barcode/qrcode reads with geolocation (gps location), thus any user ever had his/her hands on a specific code for any reason could be localized. (Imagine looking for a small but pricey item which is NOT IN THE DRAWER, but your colleague yesterday had it's barcode from the other office)

It might have privacy implications (eg. colleague home address), but it should be handled somehow by employee-legal contracts and policies. Also the phone OS might not allow location access to the app for numerous reasons.

**Reply by matmair** [reply]
Actual geo-location with sensors in the official app sounds like a very bad idea. An inventory app flies out of the window the seconds it asks me for my location and I tend to think anyone reasonable would react that way. I can understand the request but I do not see this ever making it into the official app.

**Comment by Petrox** [comment]
I do agree about the geo-consciousness on ones private phone. 
But if it is an optional feature it could be useful on company tablets, notebooks (even via website).

So while my and your preference is to disable this, there could be vehicle dealers, travel companies, farmers and forest management where geolocation could be a lifesaver and unique feature.

(also it could be VERY visible on the screen when going for a barcode/qrcode scan if GPS is enabled or not.)

**Comment by SchrodingersGat** [comment]
@Petrox if by "location" you do mean "physical / geo" location, then as @matmair says this is not something that we will be putting into the codebase. There is nothing stopping you from implementing this as a plugin if you want this behaviour.

In terms of a "location" history, this is available (automatically) for each stock item in the database. In addition to displaying the last known "location" (not geo location) it shows a transaction history for each item:

<img width="1430" alt="Screenshot 2023-08-21 at 4 45 38 pm" src="https://github.com/inventree/InvenTree/assets/10080325/9b49044d-9ae2-4de6-b346-df63069e1ec0">
tory for each item,

**Comment by Petrox** [comment]
Ok.
